### PR TITLE
Feature: Add remaining field components (Number, Date, SingleSelect, MultiSelect)

### PR DESCRIPTION
**Pull request by ahlombardini**
## Context
Building on the base field component architecture from #264, this PR adds the remaining specialized field component implementations to complete the task template field system.

## Summary
This PR introduces four additional field components:
- **NumberFieldComponent**: Numeric input with min/max constraints, decimal support, and optional unit display
- **DateFieldComponent**: Date selection using Material3 DatePicker with optional time support and date range constraints
- **SingleSelectFieldComponent**: Dropdown selection with human-readable labels, value mapping, and custom value input
- **MultiSelectFieldComponent**: Multi-selection using FilterChips with visual feedback, min/max selection constraints, and custom value support
- **Comprehensive test coverage**: All four components have complete test suites validating interaction modes, validation, constraints, and edge cases

## Why these changes
Each component builds on the BaseFieldComponent architecture established in #264, using the generic `renderer` lambda pattern to implement field-specific UI while inheriting common functionality (labels, validation, mode switching, etc.).

The implementations handle complex use cases:
- **NumberFieldComponent**: Converts string input to numeric values with proper validation and formatting
- **DateFieldComponent**: Integrates Material3 DatePicker with epoch millisecond conversion and display formatting
- **SingleSelectFieldComponent**: Uses LaunchedEffect to synchronize state when switching to edit mode, preventing stale displayed values
- **MultiSelectFieldComponent**: Provides immediate visual feedback with FilterChips, allowing users to see and modify selections without explicit save actions

## Potential future improvements
These field components are designed to be reusable and could be leveraged for all forms throughout the application beyond task templates.

## Potential future bugs
- **MultiSelectFieldComponent state synchronization**: Rapid value changes during mode transitions might cause temporary state inconsistencies due to LaunchedEffect timing

---
*Note: This PR description was generated with [Claude Code](https://claude.com/claude-code) assistance.*

### CODE REVIEW

**Review by Tuaer — CHANGES_REQUESTED** [review]
Looks very nice to me. Just a few details to change.

File: app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/NumberFieldComponent.kt [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                val trimmedText = newText.trim()
+                if (trimmedText.isEmpty()) {
+                  return@OutlinedTextField
+                }
+                if (trimmedText.matches(Regex("^-?\\d*\\.?\\d*$"))) {
+                  trimmedText.toDoubleOrNull()?.let { parsedValue ->
+                    onChange(FieldValue.NumberValue(parsedValue))
+                  }
+                }
+              },
```
[the remark this answers is not in the document]
Comment (ahlombardini) [reply app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/NumberFieldComponent.kt]: So you would suggest simply keeping the regex? and removing the .isEmpty ?
